core/packages/contracts/package.json

```diff
@@ -11,34 +11,26 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "pnpm build:sol && pnpm build:ffi",
+    "build": "pnpm build:sol && pnpm build:types",
     "build:sol": "forge build",
-    "build:ffi": "esbuild ./test/ffiWrapper.ts --bundle --platform=node --outfile=./test/ffiWrapper.js",
-    "lint": "pnpm lint:solhint && pnpm lint:eslint",
-    "lint:solhint": "solhint 'contracts/**/*.sol'",
-    "lint:eslint": "eslint .",
-    "format": "prettier --write 'contracts/**/*.sol' 'test/**/*.ts'",
-    "size": "hardhat size-contracts",
-    "coverage": "hardhat coverage --testfiles 'test/*.ts'",
+    "build:types": "typechain --target ethers-v5 'out/**/!(*.t|test).sol/!(*.abi).json' --out-dir types",
```

Comment:
Is `typechain` here nessessary in foundry stack? `forge build` will also generate abi files.

Reply:
Yes, it is very much needed. [Typechain](https://github.com/dethcrypto/TypeChain) produces Typescript bindings for our contracts. We need these in bindings for any future JS/TS libs that depend on our contracts.